### PR TITLE
bpo-43778: Fix Sphinx glossary_search extension

### PR DESCRIPTION
**Pull request by vstinner**
Create the _static/ directory if it doesn't exist.

Add also constants for the static directory and the JSON filename.

https://bugs.python.org/issue43778

### CODE REVIEW

File: Doc/tools/extensions/glossary_search.py [hunk 1]
```diff
@@ -15,6 +15,8 @@
 
 
 logger = logging.getLogger(__name__)
+STATIC_DIR = '_static'
+JSON = 'glossary.json'
```
Comment (ammaraskar) [inline Doc/tools/extensions/glossary_search.py:19]: Minor suggestion: `JSON` -> `GLOSSARY_JSON`

**Review by ammaraskar — APPROVED** [review]

File: Doc/tools/extensions/glossary_search.py (hunk 1 above)
Comment (vstinner) [reply Doc/tools/extensions/glossary_search.py:19]: Well, the extension is called "glossary_search", in the context of the extension, I don't think that it's worth it to repeat "glossary" ;-)

File: Doc/tools/extensions/glossary_search.py [hunk 2]
```diff
@@ -45,8 +47,12 @@ def on_build_finish(app, exc):
     if not app.env.glossary_terms:
         return
 
-    logger.info('Writing glossary.json', color='green')
-    with open(path.join(app.outdir, '_static', 'glossary.json'), 'w') as f:
+    logger.info(f'Writing {JSON}', color='green')
+
+    dest_dir = path.join(app.outdir, STATIC_DIR)
+    os.makedirs(dest_dir, exist_ok=True)
```
Comment (JulienPalard) [inline Doc/tools/extensions/glossary_search.py:53]: `NameError: name 'os' is not defined`

**Review by JulienPalard — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
The extension works, I didn't see https://bugs.python.org/issue43778 bug in the 3 doc CI jobs.

```
Writing glossary.json
```

The Docs PR failure on Azure is unrelated:

```
Warning, treated as error:
/home/vsts/work/1/s/Doc/using/configure.rst:416:unknown option: --enable-universalsdk
```

This issue was introduced https://bugs.python.org/issue43774 I'm aware of the issue and I will write a fix later.

File: Doc/tools/extensions/glossary_search.py (hunk 2 above)
Comment (vstinner) [reply Doc/tools/extensions/glossary_search.py:53]: Oooops, fixed ;-)

**Comment by vstinner** [comment]
I merged my PR, thanks for the reviews @ammaraskar and @JulienPalard !

**Pull request merged**